res/controllers/Arturia-KeyLab-Mk1-scripts.js

```diff
@@ -0,0 +1,759 @@
+        }
+
+        constructor(deckNumbers, midiChannel) {
+            super(deckNumbers);
+
+            // Transport buttons
+            this.playButton = new components.PlayButton({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x02],
+                type: components.Button.prototype.types.push,
+                output: delayLED(0x58),
+            });
+            this.stopButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x01],
+                type: components.Button.prototype.types.push,
+                key: "start_stop",
+                output: delayLED(0x59),
+            });
+            this.loopButton = new components.LoopToggleButton({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x37],
+                inKey: "reloop_toggle",
+                outKey: "loop_enabled",
+                output: delayLED(0x5D),
+            });
+            this.backButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x05],
+                key: "beatjump_backward",
+                output: delayLED(0x5B),
+            });
+            this.forwardButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x04],
+                key: "beatjump_forward",
+                output: delayLED(0x5C),
+            });
+
+            // Snapshot/parameter edit buttons
+            this.cueButton = new components.CueButton({
+                group: "[Channel1]",
+                midi: [0xB0 + midiChannel, 0x1F], // Snapshot 10
+                type: components.Button.prototype.types.push,
+            });
+            this.hotcues = [];
+            for (let i = 0; i < 8; i++) {
+                this.hotcues[i] = new components.HotcueButton({
+                    number: i + 1,
+                    group: "[Channel1]",
+                    midi: [0xB0 + midiChannel, 0x16 + i], // Snapshot 1-8
+                    type: components.Button.prototype.types.push,
+                    blinkState: false,
+                    blinkTimer: 0,
+                    output: function(value) {
+                        if (value === 2) {
+                            if (this.blinkTimer === 0) {
+                                this.blinkTimer = engine.beginTimer(250, () => {
+                                    this.blinkState = !this.blinkState;
+                                    delayLED(0x12 + i)(this.blinkState ? 1 : 0);
+                                });
+                            }
+                        } else {
+                            if (this.blinkTimer) {
+                                engine.stopTimer(this.blinkTimer);
+                                this.blinkTimer = 0;
+                            }
+                            delayLED(0x12 + i)(value);
+                        }
```

Comment:
can you elaborate what this does/why its needed?

Reply:
This blinks the button for the currently active loop cue so you can keep track of where in the song you are. It's not super important, but I found it to be a nice touch so that I know what loop is active and can activate the next one, jump to a previous one, or deactivate a loop without accidentally getting the wrong button.